src/Kiota.Builder/Writers/CSharp/CodeClassDeclarationWriter.cs

```diff
@@ -6,9 +6,13 @@
 namespace Kiota.Builder.Writers.CSharp;
 public class CodeClassDeclarationWriter : BaseElementWriter<ClassDeclaration, CSharpConventionService>
 {
+    private readonly string _clientClassAccessModifier;
```

Comment:
Instead of passing this over to the writer, a different approach would be to pass this to the refiner. 
https://github.com/microsoft/kiota/blob/main/src/Kiota.Builder/Refiners/CSharpRefiner.cs

The refiner would then go through all the elements of type `CodeClass` and `CodeEnum` set the accessModifier based on the configuration value. This way, we do not have to keep updating logic as more writers and code elements are introduced.